documentation/reference/relations/peers.md

```diff
@@ -0,0 +1,59 @@
+# database-peers Relation Reference Documentation
+
+This reference documentation details the implementation of the `database-peers` peer relation. This is the peer relation for PostgreSQL, used to share user and config information from the leader unit to the follower units. The file implementing these relations can be found here: [src/relations/charm.py](../../../src/relations/charm.py) (it should be moved to a file called `src/relations/peers.py` in the future).
+
+## Expected Interface
+
+These are the expected contents of the databags in this relation (all values are examples, generated in a running test instance):
+
+## Hook Handler Flowcharts
+
+These flowcharts detail the control flow of the hooks in this program. Unless otherwise stated, **a hook deferral is always followed by a return**.
+
+### Peer Relation Changed Hook
+
+```mermaid
+flowchart TD
+  hook_fired([peer-relation-changed Hook]) --> has_cluster_initialised{Has cluster\n initialised?}
+  has_cluster_initialised -- no --> defer>defer]
+  has_cluster_initialised -- yes --> is_leader{Is current\nunit leader?}
+  is_leader -- yes --> all_units_on_cluster{Are all the units \n part of the cluster?}
+  all_units_on_cluster -- yes --> is_part_of_cluster
+  all_units_on_cluster -- no --> are_all_members_ready{Are all cluster \n members ready?}
+  
+  %% This defer should have a return after it to stop the execution.
+  are_all_members_ready -- no --> defer2>defer]
+  defer2 --> is_part_of_cluster
+  
+  are_all_members_ready -- yes --> add_unit_to_cluster[Add unit to cluster]
+  add_unit_to_cluster --> patch_pod_labels[Patch pod labels of the new cluster member]
+  patch_pod_labels --> all_units_on_cluster
+  is_leader -- no --> is_part_of_cluster{Is current unit \n part of the cluster?}
```

Comment:
I believe if you move this up to line 20, you might have a slightly cleaner line in your flowchart

Reply:
That's right. Thanks for the tip!